app/controllers/publishers/ats_api/v1/create_vacancy_service.rb

```diff
@@ -0,0 +1,66 @@
+module Publishers
+  module AtsApi
+    module V1
+      class CreateVacancyService
+        extend OrganisationFetcher
+
+        InvalidOrganisationError = Class.new(StandardError)
+
+        class << self
+          def call(params)
+            vacancy = Vacancy.new(sanitised_params(params))
+
+            if (conflict = conflict_vacancy(vacancy))
+              return conflict_response(conflict)
+            end
+
+            if vacancy.save
+              success_response(vacancy)
+            else
+              validation_error_response(vacancy)
+            end
+          end
+
+          private
+
+          def sanitised_params(params)
+            organisations = fetch_organisations(params[:schools])
+            raise InvalidOrganisationError, "No valid organisations found" if organisations.blank?
+
+            params[:publish_on] ||= Time.zone.today.to_s
+            params.except(:schools).merge(organisations: organisations)
+          end
+
+          def conflict_vacancy(vacancy)
+            Vacancy.find_by(external_reference: vacancy.external_reference)
```

Comment:
There is a chance (little, but possible) that multiple clients use the same `external_reference` within their internal systems. 
EG: 1, 2, 3...
We should also use the `publisher_ats_api_client` combined with the `external_reference` to ensure that we find a conflicting vacancy for the current client.